imports/plugins/core/orders/client/components/lineItems.js

```diff
@@ -57,91 +78,233 @@ class LineItems extends Component {
-                  <div key={i}>
-                    { this.renderLineItem(uniqueItem, 1) }
-                    { this.renderLineItemInvoice(uniqueItem) }
-                  </div>
-                )}
+  renderPopOver() {
+    return (
+      <Popover
+        isOpen={this.props.popOverIsOpen}
+        attachment="middle center"
+        targetAttachment="middle center"
+        showDropdownButton={false}
+      >
+        {this.popOverContent()}
+      </Popover>
+    );
+  }
 
-              </div>
+  popOverContent() {
+    return (
+      <div className="invoice-popover">
+        <div className="invoice-popover-controls">
+          <Checkbox
+            className="checkbox-large"
+            checked={this.props.selectAllItems || this.props.selectedItems.length === this.props.uniqueItems.length}
+            onChange={() => this.props.handleSelectAllItems(this.props.uniqueItems)}
+          />
+          <div className="invoice-popover-close">
+            <Button
+              className="rui btn btn-default flat icon-only pull-right"
+              icon="fa-2x fa fa-times"
+              bezelStyle="flat"
+              onClick={this.props.clearRefunds}
+            />
+          </div>
+        </div>
+        <div>
+          {this.props.uniqueItems.map((uniqueItem, index) => (
+            <div key={index}>
+              {this.renderPopOverLineItem(uniqueItem)}
+              {this.renderLineItemInvoice(uniqueItem)}
             </div>
+          ))}
+        </div>
+        <div>
+          {!isEmpty(this.props.editedItems) && this.renderLineItemRefund()}
+        </div>
+        <div className="invoice-actions">
+          <div className="invoice-action-cancel">
+            <Button
+              className="pull-right"
+              bezelStyle="solid"
+              status="default"
+              label="Cancel"
+              onClick={this.props.clearRefunds}
+            />
+          </div>
+          <div className="invoice-action-refund">
+            <Button
+              className="pull-right"
+              bezelStyle="solid"
+              status="primary"
+              disabled={this.props.isRefunding || this.props.editedItems.length === 0}
+              onClick={this.props.handleReturnItems}
+            >
+              {this.props.isRefunding ? <span>Refunding <i className="fa fa-spinner fa-spin" /></span> :
+                <span>Refund Items</span>
+              }
+            </Button>
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  render() {
+    const { uniqueItems } = this.props;
+    return (
+      <div className="invoice invoice-line-items" onClick={this.props.handlePopOverOpen}>
+        {uniqueItems.map((uniqueItem) => {
+          return (
+            <div key={uniqueItem._id}> {this.renderLineItem(uniqueItem)} </div>
           );
         })}
+
+        {
+          Roles.userIsInRole(Meteor.userId(), ["orders", "dashboard/orders"], Reaction.getShopId()) &&
```

Comment:
@spencern Should this is a `hasPermission` call now?

Reply:
@joykare @zenweasel `all` instances of `Roles.userIsInRole` should be deprecated, removed and use `hasPermissions` now.

Reply:
See #2800